sqlglot/dialects/teradata.py

```diff
@@ -161,3 +161,13 @@ def rangen_sql(self, expression: exp.RangeN) -> str:
             each_sql = f" EACH {each_sql}" if each_sql else ""
 
             return f"RANGE_N({this} BETWEEN {expressions_sql}{each_sql})"
+
+        def cast_sql(self, expression: exp.Cast) -> str:
```

Comment:
delete all of this it's not needed, you only need to modify the base def datatype_sql

Reply:
datatype_sql already has "nested" so add format there

Reply:
the idea is for the generic dialect to be very powerful. how else would you represent it?